### PR TITLE
[Portfolio] - the list of incoming changes:

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.monday.com/boards/2218344956/pulses/2426803546?doc_id=2603614244&userId=28299221

AC:
 - Rename WalletPage -> PortfolioPage
 - Rename the tab "Vested Assets" to "Vesting"
 - Rename "NFT" to "Gallery"
 - Remove the "Action" label - the buttons next to each asset shows "Send" and "Receive"
 - Add a hint when hovering (or clicking) on the buttons "Send" and "Receive" with text "Send / Receive [selected asset] to / from your connected address in the RSK network"
 - Remove the middle "Swap history" section
 - Rename "Vested history" to "Vesting history"
 - Added "light" refactoring of pages/components that have been changed as well as relative components

### CODE REVIEW

**Comment by tiltom** [comment]
1. `Portfolio` and `Rewards` should be aligned with the table.
2. Discuss the inactive button color with Gilad because there is a darker background in the design so the contrast is ok but it's not very good with our current background so we need to adjust it (and we don't want to change the background as we want to have it consistent).
3. I think that `Connected Wallet` text is redundant, discuss it with Gilad as well.
4. Discuss `transak` flow (previously under `Buy` button in `RBTC`) with Pili.
Screenshot 2022-05-27 at 10 03 44

**Comment by tiltom** [comment]
If I click on `Rewards`, the `Portfolio` button is really hard to see on the background image, discuss it with Gilad pls (similar to point no.2 in my previous comment).
image

File: src/app/containers/VestedHistory/components/VestedHistoryTable.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        return (
+          <VestedHistoryTableRow
+            key={item.transactionHash + index}
+            item={item}
+          />
+        );
+      })}
+    </>
+  );
+};
```
Comment (tiltom) [inline src/app/containers/VestedHistory/components/VestedHistoryTable.tsx:23]:
```suggestion
export const VestedHistoryTable: React.FC<IVestedHistoryTable> = ({
  items,
}) => (
  <>
    {items.map((item, index) => (
      <VestedHistoryTableRow key={item.transactionHash + index} item={item} />
    ))}
  </>
);
```

Comment (tiltom) [reply src/app/containers/VestedHistory/components/VestedHistoryTable.tsx:23]: Btw do you need to have a composite key (`item.transactionHash + index`) ?

File: src/app/containers/VestedHistory/components/VestedHistoryTableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          <div>
+            {!item.status && (
+              <p className="tw-m-0">{t(translations.common.confirmed)}</p>
+            )}
+            {item.status === TxStatus.FAILED && (
+              <p className="tw-m-0">{t(translations.common.failed)}</p>
+            )}
+            {item.status === TxStatus.PENDING && (
+              <p className="tw-m-0">{t(translations.common.pending)}</p>
+            )}
```
Comment (tiltom) [inline src/app/containers/VestedHistory/components/VestedHistoryTableRow.tsx:78]: Could you extract it to a function (e.g. `getStatusText`, can be in this file)?

File: src/app/containers/VestedHistory/components/VestedHistoryTableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          <div>
+            {!item.status && (
+              <Icon icon="success-tx" size={25} className="tw-text-success" />
+            )}
+            {item.status === TxStatus.FAILED && (
+              <Icon icon="failed-tx" size={25} className="tw-text-warning" />
+            )}
+            {item.status === TxStatus.PENDING && (
+              <Icon icon="pending-tx" size={25} className="tw-text-sov-white" />
+            )}
```
Comment (tiltom) [inline src/app/containers/VestedHistory/components/VestedHistoryTableRow.tsx:93]: Extract it to a function pls (e.g. `getStatusIcon`).

File: src/app/pages/PortfolioPage/index.tsx [hunk 4]
```diff
@@ -27,62 +24,68 @@ export const WalletPage: React.FC = () => {
...
-    },
     {
       id: 'vestedHistory',
       label: t(translations.vestedHistory.title),
       content: <VestedHistory />,
     },
   ];
 
-  const walletTabs = [
+  const portfolioTabs = [
```
Comment (tiltom) [inline src/app/pages/PortfolioPage/index.tsx:34]: Use `useMemo` (for `historyTabs` as well).

File: src/app/pages/PortfolioPage/types.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export enum PortfolioDirectionType {
```
Comment (tiltom) [inline src/app/pages/PortfolioPage/types.ts:1]:
```suggestion
export enum HistoryItemType {
```

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/containers/VestedHistory/components/VestedHistoryTable.tsx (hunk 1 above)
Comment (pietro-maximoff) [reply src/app/containers/VestedHistory/components/VestedHistoryTable.tsx:23]: I think the index will be enough

**Review by tiltom — DISMISSED** [review]

**Review by creed-victor — DISMISSED** [review]

**Comment by jjmr007** [comment]
### My Observations:

> https://sovryn.monday.com/boards/2218344956/pulses/2426803546?doc_id=2603614244&userId=28299221
> 
> AC:
> 
> * Rename WalletPage -> PortfolioPage --->  **The rename was well done, but I don't see any direct link to the 'portfolio' in the menu bar**
> * Rename the tab "Vested Assets" to "Vesting"  ---> **menu "Earn" -> "Staking": the change in the name was done; in "Portfolio" besides the "Assets" menu, we see "Vesting", change in the name was done**
> * Rename "NFT" to "Gallery" ---> **In "Portfolio" we still see the menu "NFT" instead of "Gallery"... MISSING CHANGE**
> * Remove the "Action" label - the buttons next to each asset shows "Send" and "Receive" ---> **"Action" label removed **
> * Add a hint when hovering (or clicking) on the buttons "Send" and "Receive" with text "Send / Receive [selected asset] to / from your connected address in the RSK network" ---> **MISSPELLED: When we try to send we send _from_ my connected address, when we receive, we receive _to_, the order is inverted**
> * Remove the middle "Swap history" section ---> **In the menu bar -> "Trade" menu -> "Swap" menu -> we can still see the "Swap history"... it appears and dissapears intermittently as you can see in this [video](https://drive.google.com/file/d/1QiKpLJAhEd54g63SkCoMYrAoNTOPL3By/view?usp=sharing)**
> * Rename "Vested history" to "Vesting history" ---> **where is supposed to be done this change? in the menu "Earn" -> "Staking" there no "Vested History" whether in the original or this PR UI, there is only "Staking History"**
> * Added "light" refactoring of pages/components that have been changed as well as relative components

**Review by soulBit — CHANGES_REQUESTED** [review]
Changes requested via Monday.com

**Review by soulBit — APPROVED** [review]

**Review by Alitasovryn — APPROVED** [review]

**Pull request merged**